### ISSUE TITLE
Remove --tls-verify=false from the pipeline steps

### ISSUE THREAD

**Issue by kvijai82** [kabanero-io/kabanero-pipelines#129]
Add the CA cert into the trust store instead like we do in the assemble-extract step.

**Comment by aadeshpa** [comment]
After I had discussion with @kvijai82 below is the current behavior and we are planning to change it as below

Current behavior of build-push task:
1. In the build-push task in the build step , for internal image registry `image-registry.openshift-image-registry.svc:5000` we create openssl certificate since it gives certificate error if we do not.
2. In same build step under the task, if user provides private image registry then too we create openssl certificate to solve the certificate error.
3. For all the public docker.io and other docker registry we skip the certificate creation since it does not give error and the pipeline gets stuck for 5 to 7 min at the ssl certificate creation logic.
4. Along with `build` step in build-push task, the `push` step uses `buildah push` command and we currently we use `--tls-verify=false` option which avoids ssl creation logic . Also in `image-scan` task there are two locations where we have the `--tls-verify=false` option used in the buildah commands. 

Planned behavior with this change:
1. In build-push task, under the build step we plan to add the internal image registry to `/etc/container/registries.conf` file as `[registries.insecure]` so certificate creation for that piece is not required.
2. For all the private image registries we would continue creating the ssl certificate.
3. For all public docker registries like docker.io and others we will continue skipping the ssl creation logic.
4. Along with `build` step in build-push task, the `push` step uses `buildah push` command and we will plan to create ssl certificate for private registries . Also in `image-scan` task there are two locations where we will call the ssl creation logic and remove the `--tls-verify=false` from the buildah commands.

**Comment by aadeshpa** [comment]
As per @jgawor@us.ibm.com:
@kalathur as to the whole certificate issue (doing openssl stuff), i think a simpler solution is to update the /etc/container/registries.conf file in the pod and add a given registry to the insecure list. Something like

INSECURE_REGISTRTY="..."
sed -i "0,/registries = \[\]/s/registries = \[\]/registries = \[$INSECURE_REGISTRY'\]/" /etc/containers/registries.conf

Also, it would be nice to inject a whole list of insecure registries (instead of being limited to 1) and even better to get that list from image.config.openshift.io/cluster CR.

Basically, for ICPA 4.2 we should aim to remove this whole ConfigMap work-around.

**Comment by aadeshpa** [comment]
We figured out how to get the internal and external route url of the internal registry.
```
external_route_internal_registry=$(kubectl get image.config.openshift.io/cluster -o yaml --output="jsonpath={.status.externalRegistryHostnames[0]}")
             internal_route_internal_registry=$(kubectl get image.config.openshift.io/cluster -o yaml --output="jsonpath={.status.internalRegistryHostname}")

```

**Comment by kvijai82** [comment]
Doc issue: https://github.com/kabanero-io/docs/issues/373

**Comment by kvijai82** [comment]
Fix merged in 0.8.0 release.
